### ISSUE TITLE
Allow To Instantly Unload/Reload Extensions In Game

### ISSUE THREAD

**Issue by BrownieGamingX5** [absoIute/Mega-Hack-Pro-Future#612]
Maybe there could be a feature of where you can instantly unload and reload extensions in game. This might seem a bit vague, but I will give more information on this:

- [ ] **Unloading An Extension: When you unload an extension, it should happen instantly, and you should see the sprites of the extension disappear instantly too.**

- [ ] **Loading An Extension: When you load the extension back, the sprites from that extension should pop back up instantly as well.**

This could be useful for people who do not want to click "Refresh Textures" or reload the game, just so the sprites from the extensions could disappear, waiting for you to reload it. This could also be used for the removal of another feature: "Load First" and "Load Last". Instead, they could be replaced with "Load" and "Unload".

**Comment by Nicko2102** [comment]
> This could also be used for the removal of another feature: "Load First" and "Load Last". Instead, they could be replaced with "Load" and "Unload".

IIRC, the "load first" and "load last" feature is being replaced by a drag+drop system, and i don't think people with 10 extensions will want to click the load button on all of them every time they reopen the game, but if the other ideas are possible i would love to see it implemented.

The only issue is that some extensions are not just textures, but functionalities as well. For example, trying to unload betteredit while in the betteredit settings menu might cause some problems.

**Comment by defaultacc30** [comment]
> > This could also be used for the removal of another feature: "Load First" and "Load Last". Instead, they could be replaced with "Load" and "Unload".
> 
> IIRC, the "load first" and "load last" feature is being replaced by a drag+drop system, and i don't think people with 10 extensions will want to click the load button on all of them every time they reopen the game, but if the other ideas are possible i would love to see it implemented.
> 
> The only issue is that some extensions are not just textures, but functionalities as well. For example, trying to unload betteredit while in the betteredit settings menu might cause some problems.

well actually, absolute is switching to geode which has the sexiest ui ever.

**Comment by Nicko2102** [comment]
unrelated

**Comment by defaultacc30** [comment]
https://twitter.com/FigmentBoy_/status/1493029613487935489

**Comment by defaultacc30** [comment]
> your comment has nothing to do with what i said about funcionalities in current extensions being unable to be unloaded while in use. also im not clicking your ip grabber.

HOW IS IT A IP GRABBER

**Comment by defaultacc30** [comment]
> your comment has nothing to do with what i said about funcionalities in current extensions being unable to be unloaded while in use. also im not clicking your ip grabber.

absolute is switching to geode, it is up to the devs of geode if they want a customizer for loading. you can suggest in the gd moddings btw if your wondering

**Comment by absoIute** [comment]
> > > This could also be used for the removal of another feature: "Load First" and "Load Last". Instead, they could be replaced with "Load" and "Unload".
> > 
> > 
> > IIRC, the "load first" and "load last" feature is being replaced by a drag+drop system, and i don't think people with 10 extensions will want to click the load button on all of them every time they reopen the game, but if the other ideas are possible i would love to see it implemented.
> > The only issue is that some extensions are not just textures, but functionalities as well. For example, trying to unload betteredit while in the betteredit settings menu might cause some problems.
> 
> well actually, absolute is switching to geode which has the sexiest ui ever.

Geode is not out yet, I will not be exclusively switching to Geode either as of yet

**Comment by absoIute** [comment]
Unfortunately I've decided to not go down this route since it expects all extensions to be able to unloaded (most devs don't account for this so it just leads to crashes instead).

Same with loading, many mods are designed to be loaded on launch which is why I've required restarting every time an extension is loaded.

**Comment by defaultacc30** [comment]
> Unfortunately I've decided to not go down this route since it expects all extensions to be able to unloaded (most devs don't account for this so it just leads to crashes instead).
> 
> Same with loading, many mods are designed to be loaded on launch which is why I've required restarting every time an extension is loaded.

so are you not going to switch to geode?

**Comment by Nicko2102** [comment]
> > Unfortunately I've decided to not go down this route since it expects all extensions to be able to unloaded (most devs don't account for this so it just leads to crashes instead).
> > Same with loading, many mods are designed to be loaded on launch which is why I've required restarting every time an extension is loaded.
> 
> so are you not going to switch to geode?

to quote absolute, "I will not be exclusively switching to Geode as of yet"

**Comment by defaultacc30** [comment]
> ^ in layman's terms that means I will be releasing Geode & non-Geode versions

well, i feel like thats a bit redudant. maybe a switcher?

**Comment by Nicko2102** [comment]
Absolute's choice, users can download whichever they want and/or switch between them as they wish

**Comment by defaultacc30** [comment]
> Absolute's choice, users can download whichever they want and/or switch between them as they wish

they can confuse it for mhv6 download.

**Comment by Nicko2102** [comment]
> > Absolute's choice, users can download whichever they want and/or switch between them as they wish
> 
> they can confuse it for mhv6 download.

i doubt anyone has confused MHV7 with MHV6, so why would someone confuse MHV8 with MHV6?

**Comment by absoIute** [comment]
It's going to be done in a way that won't require the user to know what Geode is, it'll just install the correct version

**Comment by Nicko2102** [comment]
> no like mhv7 without geode and with geode people wont know what geode is so he needs to make a explantion

thats not at all what you said before lmao
